keyvalue/client-internal/src/main/java/tachyon/client/keyvalue/KeyValueStores.java

```diff
@@ -38,14 +38,24 @@
     private Factory() {} // to prevent initialization
 
     /**
-     * @return a {@link KeyValueStores} instance
+     * @return a (cached) {@link KeyValueStores} instance
      */
     public static synchronized KeyValueStores create() {
       if (sKeyValueStores == null) {
-        sKeyValueStores = new BaseKeyValueStores();
+        reset();
       }
       return sKeyValueStores;
     }
+
+    /**
+     * {@link tachyon.client.ClientContext} may be reset in different tests running in the same JVM,
+     * in this case, the cached {@link KeyValueStores} needs to be updated.
+     *
+     * @return a newly created {@link KeyValueStores}
+     */
+    public static synchronized void reset() {
```

Comment:
If key-value master client in `BaseKeyValueStores` is created on demand, this method does not need to exist, but there is no reason not to use a shared master client in that class. 

Could you come up with a better idea?